cmd/gossamer/main.go

```diff
@@ -425,7 +425,7 @@ func pruneState(ctx *cli.Context) error {
 	}
 
 	bloomSize := ctx.GlobalUint64(BloomFilterSizeFlag.Name)
-	retainBlocks := ctx.GlobalInt64(RetainBlockNumberFlag.Name)
+	retainBlocks := uint32(ctx.GlobalUint64(RetainBlockNumberFlag.Name))
```

Comment:
We should add a check that the value `ctx.GlobalUint64(RetainBlockNumberFlag.Name)` is less than the max uint32 (`^uint32(0)`)